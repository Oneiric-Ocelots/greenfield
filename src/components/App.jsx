import React, { useState, useEffect } from "react";
import ProductDetailsContainer from "../containers/ProductDetailsContainer.js";
import RelatedItems from "./RelatedItemsComponents/RelatedItems.jsx";
import ReviewsContainer from "../containers/RatingsReviewsContainers/ReviewsContainer.js";
import QAContainer from "../containers/QAContainer.js";
import { makeStyles } from "@material-ui/core/styles";
import Paper from "@material-ui/core/Paper";
import Grid from "@material-ui/core/Grid";

const App = props => {
  return (
    <div className="components">
      <Grid container spacing={4}>
        <Grid item xs={12}>
          <ProductDetailsContainer />
        </Grid>
        <Grid item xs={11}>
          <RelatedItems />
        </Grid>
        <Grid item xs={11}>
          <QAContainer />
        </Grid>
        <Grid item xs={11}>
          <ReviewsContainer />
        </Grid>
      </Grid>
    </div>
  );
};

export default App;
