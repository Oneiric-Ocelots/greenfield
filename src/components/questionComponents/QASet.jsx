import React, { Component } from "react";
import axios from "axios";
import Answers from "./Answers.jsx";
import AddAnswer from "./AddAnswer.jsx";
import MoreAnswers from "./MoreAnswers.jsx";

export default class QASet extends Component {
  constructor(props) {
    super(props);
    this.state = {
      answers: [],
      counter: 2,
      qHelpful: false
    };
    this.getAnswers = this.getAnswers.bind(this);
    this.showMoreAnswers = this.showMoreAnswers.bind(this);
  }
  componentDidMount() {
    this.getAnswers(this.props.question.question_id);
  }

  getAnswers() {
    axios
      .get(
        `http://18.223.1.30/qa/${this.props.question.question_id}/answers/?count=1000`
      )
      .then(({ data }) => {
        this.setState({ answers: data.results });
      });
  }
  showMoreAnswers() {
    this.setState({ counter: this.state.counter + 2 });
  }
  markQuestionHelpful() {
    axios
      .put(
        `http://18.223.1.30/qa/question/${this.props.question.question_id}/helpful`
      )
      .then(() => {
        this.props.getQuestions(this.props.product.id);
      })
      .catch(err => {
        console.log(err);
      });
  }
  render() {
    let moreAnswersBtn;
    if (this.state.counter >= this.state.answers.length) {
      moreAnswersBtn = null;
    } else {
      moreAnswersBtn = <MoreAnswers />;
    }
    return (
      <div className="q-a-set">
        <div>Q: {this.props.question.question_body}</div>
        <div className="q-a-set-right">
          Helpful? &nbsp;
          <span
            className="yes-button"
            onClick={() => {
              if (!this.state.qHelpful) {
                this.setState({ qHelpful: true });
                this.markQuestionHelpful();
              }
            }}
          >
            Yes
          </span>
          {" ("} {this.props.question.question_helpfulness}
          {")"} |
          <AddAnswer
            getAnswers={this.getAnswers}
            product={this.props.product}
            qbody={this.props.question.question_body}
            questionId={this.props.question.question_id}
          />
        </div>
        <div>
          <div>A: </div>
          {this.state.answers.slice(0, this.state.counter).map(answer => {
            return (
              <Answers
                getAnswers={this.getAnswers}
                key={answer.answer_id}
                answer={answer}
              />
            );
          })}
          {this.state.counter >= this.state.answers.length ? null : (
            <MoreAnswers showMoreAnswers={this.showMoreAnswers} />
          )}
        </div>
      </div>
    );
  }
}
