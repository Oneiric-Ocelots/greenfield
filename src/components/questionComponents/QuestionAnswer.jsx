import React from "react";
import SearchQuestions from "./SearchQuestions.jsx";
import QASet from "./QASet.jsx";

export default function QuestionAnswer() {
  return (
    <div className="q-and-a">
      QUESTIONS & ANSWERS
      <SearchQuestions />
      <QASet />
      more question button add question button
    </div>
  );
}
