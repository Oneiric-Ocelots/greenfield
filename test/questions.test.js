import React from "react";
import { mount, shallow, configure } from "enzyme";
// import Adapter from "enzyme-adapter-react-16";
// import "babel-polyfill";
// configure({ adapter: new Adapter() });

import QuestionAnswer from "../src/components/questionComponents/QuestionAnswer.jsx";
import QASet from "../src/components/questionComponents/QASet.jsx";
import App from "../src/components/App.jsx";

// let samplet2 = (a, b) => {
//   return a + b;
// };

// test("adds 1+2 to equal to 3", () => {
//   expect(samplet2(1, 5)).toBe(6);
// });

// describe("Questions", () => {
//   it("should render max 2 questions on load", () => {
//     const wrapper = shallow(<App productInfo={{ id: 1 }} />);
//     expect(wrapper)
//       .exists()
//       .to.equal(true);
//   });

//   // it("render correctly text component", () => {
//   //   const QAComponent = renderer.create(<QuestionAnswer />).toJSON();
//   //   expect(QAComponent).toMatchSnapshot();
//   // });
// });

test("Questions test", () => {
  expect(
    shallow(<QuestionAnswer productId={{ id: 1 }} />).contains(
      <div className="q-and-a">
        QUESTIONS & ANSWERS
        <SearchQuestions
          questions={questions}
          updateSearchQs={updateSearchQs}
          setQuestions={setQuestions}
          getQuestions={getQuestions}
          productId={props.productInfo.id}
        />
      </div>
    )
  ).toBe(true);
});
