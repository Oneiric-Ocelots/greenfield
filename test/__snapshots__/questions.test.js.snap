// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test that QA Component is connected to store QA Component renders 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <QuestionAnswer
    props={
      Object {
        "productInfo": Object {
          "id": 1,
        },
      }
    }
  />
</Provider>
`;
